docs/src/design/chiplets/main.md

```diff
@@ -172,18 +175,28 @@ Lookup requests are sent to the chiplets bus by the following components:
 - The stack sends requests for [bitwise](../stack/u32_ops.md#u32and), [memory](../stack/io_ops.md#memory-access-operations), and [cryptographic hash operations](../stack/crypto_ops.md).
 - The decoder sends requests for [hash operations](../decoder/main.md#program-block-hashing) for program block hashing.
 - The decoder sends a procedure access request to the [Kernel ROM chiplet](./kernel_rom.md) for each `SYSCALL` during [program block hashing](../decoder/main.md#program-block-hashing).
+- The verifier initializes the bus with requests to the [Kernel ROM chiplet](./kernel_rom.md) for each unique kernel procedure hash.
 
 Responses are provided by the [hash](./hasher.md#chiplets-bus-constraints), [bitwise](./bitwise.md#chiplets-bus-constraints), [memory](./memory.md#chiplets-bus-constraints), and [kernel ROM](./kernel_rom.md#chiplets-bus-constraints) chiplets.
 
+The chiplet bus can be initialized with reduced request $v_0, v_1, \ldots$ by computing their product $v_{init} = \prod_i v_i$, and enforcing the boundary constraint in the first row to ensure 
+$b_{chip} = \frac{1}{v_{init}}$ 
+
+> $$
+b_{chip} \cdot v_{init} - 1 = 0 \text{ | degree} = 2
+$$
+
 ## Chiplets virtual table
 
 Some chiplets require the use of a [virtual table](../lookups/multiset.md#virtual-tables) to maintain and enforce the correctness of their internal state. Because the length of these virtual tables does not exceed the length of the chiplets themselves, a single virtual table called $vt_{chip}$ can be shared by all chiplets.
 
 Currently, the chiplets virtual table combines two virtual tables:
 - the hash chiplet's [sibling table](./hasher.md#sibling-table-constraints)
-- the kernel ROM chiplet's [kernel procedure table](./kernel_rom.md#kernel-procedure-table-constraints)
+- TODO: a shared table between the [ACE](./ace.md#chiplets-bus-constraints) and [memory](./memory.md#chiplets-bus-constraints) chiplets enabling the ACE chiplet to read inputs and the circuit from memory.
 
-To combine these correctly, the [running product column](../lookups/multiset.md) for this table must be constrained not only at the beginning and the end of the trace, but also where the hash chiplet ends and where the kernel ROM chiplet begins. These positions can be identified using the chiplet selector columns.
+To combine these correctly, the [running product column](../lookups/multiset.md) for this table must be constrained not only at the beginning and the end of the trace, but also where the hash chiplet ends. 
+Other chiplets can use the virtual table to communicate amongst each other, for example, to enable a chiplet to access the memory.
+Since the hasher chiplet is in the first position, and running product columns can only be initialized in its first row, the table must be empty at the end of the chiplets trace.
```

Comment:
I thin we should emphasize the separation between a bus and a virtual table. Basically the bus used to implement the hasher sibling vtable is also used to route read requests from ACE to Memory chiplet